components/bme280/src/bme280.c

```diff
@@ -0,0 +1,309 @@
+    {
+        ESP_LOGI(TAG_BME280, "BME280 set oversamp success");
+        return ESP_OK;
+    }
+    else
+    {
+        ESP_LOGE(TAG_BME280, "BME280 set oversamp failed. code: %d", com_rslt);
+        return ESP_FAIL;
+    }
+}
+
+esp_err_t bme280_set_settings(uint8_t standby_time, uint8_t filter_coeff, uint8_t power_mode)
+{
+    // Result of communication results
+    s32 com_rslt;
+
+    // Map settings to actual values for the driver
+    uint8_t standby_settings[8] = {
+        BME280_STANDBY_TIME_1_MS,
+        BME280_STANDBY_TIME_63_MS,
+        BME280_STANDBY_TIME_125_MS,
+        BME280_STANDBY_TIME_250_MS,
+        BME280_STANDBY_TIME_500_MS,
+        BME280_STANDBY_TIME_1000_MS,
+        BME280_STANDBY_TIME_10_MS,
+        BME280_STANDBY_TIME_20_MS};
+
+    uint8_t filter_settings[5] = {
+        BME280_FILTER_COEFF_OFF,
+        BME280_FILTER_COEFF_2,
+        BME280_FILTER_COEFF_4,
+        BME280_FILTER_COEFF_8,
+        BME280_FILTER_COEFF_16};
+
+    uint8_t power_settings[3] = {
+        BME280_SLEEP_MODE,
+        BME280_FORCED_MODE,
+        BME280_NORMAL_MODE};
+
+    standby_time = standby_settings[standby_time];
+    filter_coeff = filter_settings[filter_coeff];
+    power_mode = power_settings[power_mode];
+
+    // Set the settings
+    com_rslt = bme280_set_standby_durn(standby_time);
+    com_rslt += bme280_set_filter(filter_coeff);
+    com_rslt += bme280_set_power_mode(power_mode);
+
+    // Check if the settings were set correctly
+    if (com_rslt == SUCCESS)
+    {
+
+        ESP_LOGI(TAG_BME280, "BME280 set settings success");
+        return ESP_OK;
+    }
+    else
+    {
+        ESP_LOGE(TAG_BME280, "BME280 set settings failed. code: %d", com_rslt);
+        return ESP_FAIL;
+    }
+}
+
+double bme280_read_pressure(void)
+{
+    s32 com_rslt;
+
+    // Uncompenstated pressure value
+    s32 v_uncomp_pressure_s32;
+
+    // Read the uncompensated pressure value
+    com_rslt = bme280_read_uncomp_pressure(&v_uncomp_pressure_s32);
+
+    // Check if the read was successful
+    if (com_rslt == SUCCESS)
+    {
+        // Compensate the pressure value and return it
+        return bme280_compensate_pressure_double(v_uncomp_pressure_s32);
+    }
+    else
+    {
+        ESP_LOGE(TAG_BME280, "BME280 uncomp pressure read failed. code: %d", com_rslt);
+        return 0;
+    }
+}
+
+double bme280_read_temperature(void)
+{
+    s32 com_rslt;
+
+    // Uncompenstated temperature value
+    s32 v_uncomp_temperature_s32;
+
+    // Read the uncompensated temperature value
+    com_rslt = bme280_read_uncomp_temperature(&v_uncomp_temperature_s32);
+
+    // Check if the read was successful
+    if (com_rslt == SUCCESS)
+    {
+        // Compensate the temperature value and return it
+        return bme280_compensate_temperature_double(v_uncomp_temperature_s32);
+    }
+    else
+    {
+        ESP_LOGE(TAG_BME280, "BME280 uncomp temperature read failed. code: %d", com_rslt);
+        return 0;
```

Comment:
But what in case, if temperature is 0C? How do you know if this is a freezing temperature reading or error code?

Consider returning a status code by return and temperature by parameter sth. like:
```c
reading_status_t bme280_read_temperature(double *reading);
```